adafruit_requests.py

```diff
@@ -555,9 +567,14 @@ def _send_as_bytes(self, socket: SocketType, data: str):
         return self._send(socket, bytes(data, "utf-8"))
 
     def _send_header(self, socket, header, value):
+        if value is None:
```

Comment:
In `CPython requests` they have a method [merge_setting](https://github.com/psf/requests/blob/96b22fa18c00831656ee4b286bf1c9062459b00a/src/requests/sessions.py#L61)

That they use in a bunch of places, but it creates new dict objects, and thought this would be simpler in memory usage